Fix notification payment lookup (#1049)
commercetools/commercetools-adyen-integration: docs/adr/0011-matching-adyen-notification.md

Since [v9.10.0](https://github.com/commercetools/commercetools-adyen-integration/releases/tag/v9.10.0), we have introduced custom reference for refund.
Payment key could be the custom field in payment transaction defined by user, therefore payment is no longer always able to be obtained by merchant reference as key. 

The alternative for that is to use `pspReference` as payment `key` field.

## Decision

- We will use `originalReference` (or `pspReference` if `originalReference` does not exist) or `merchantReference` as payment key for matching payment for notification.
- For web component version 4, `pspReference` can be obtained from `makePaymentResponse` in extension module. It is used to update as payment key. Therefore we can match the payment by `pspReference` given in notification.
- For web component version 5, `pspReference` is first provided in notification with AUTHORIZATION event. It is different from web component version 4, in which `pspReference` has already been provided from Adyen API response in extension module, and used to update as payment key. Therefore we still need `merchantReference` for payment lookup in this scenario.

> Once payment can be matched, update the payment key with `pspReference` obtained in notification.

Reviewer: ```suggestion
Once payment is found, update the payment key with `pspReference` obtained in notification.
```
